pyanaconda/ui/gui/spokes/custom_storage.py

```diff
@@ -763,331 +702,341 @@ def _save_right_side(self, selector):
-                changed_container = True
-
-        old_device_info["container_name"] = old_container_name
-        new_device_info["container_name"] = container_name
-
-        container_encrypted = self._device_container_encrypted
-        old_device_info["container_encrypted"] = old_container_encrypted
-        new_device_info["container_encrypted"] = container_encrypted
-        changed_container_encrypted = (container_encrypted != old_container_encrypted)
-
-        container_raid_level = self._device_container_raid_level
-        if container_raid_level not in get_supported_container_raid_levels(device_type):
-            container_raid_level = get_default_container_raid_level(device_type)
-
-        old_device_info["container_raid_level"] = old_container_raid_level
-        new_device_info["container_raid_level"] = container_raid_level
-        changed_container_raid_level = (old_container_raid_level != container_raid_level)
-
-        container_size = self._device_container_size
-        old_device_info["container_size"] = old_container_size
-        new_device_info["container_size"] = container_size
-        changed_container_size = (old_container_size != container_size)
-
-        # DISK SET
-        old_disks = device.disks
-        if hasattr(device, "req_disks") and not device.exists:
-            old_disks = device.req_disks
-
-        disks = self._device_disks[:]
-        if container and changed_device_type:
+
+        if container and old_device_info["device_type"] != new_device_info["device_type"]:
             log.debug("overriding disk set with container's")
-            disks = container.disks[:]
-        changed_disk_set = (set(old_disks) != set(disks))
-        old_device_info["disks"] = old_disks
-        new_device_info["disks"] = disks
-        log.debug("old disks: %s", [d.name for d in old_disks])
-        log.debug("new disks: %s", [d.name for d in disks])
-
-        log.debug("device: %s", device)
-        already_logged = {"disks", "device"}
-        # log the other changes (old_device_info doesn't have the 'device' key)
-        for key in (to_log for to_log in
-                    old_device_info.keys() if to_log not in already_logged):
-            log.debug("old %s: %s", key, old_device_info[key])
-            log.debug("new %s: %s", key, new_device_info[key])
-
-        # XXX prevent multiple raid or encryption layers?
-
-        changed = (changed_name or changed_size or changed_device_type or
-                   changed_label or changed_mountpoint or changed_disk_set or
-                   changed_encryption or changed_luks_version or
-                   changed_raid_level or changed_fs_type or
-                   changed_container or changed_container_encrypted or
-                   changed_container_raid_level or changed_container_size)
+            new_device_info["disks"] = container.disks[:]
+
+    def _get_device_info_description(self, new_device_info, old_device_info):
```

Comment:
It should be probably `_get_new_device_info_description`.

Reply:
Fixed.